### PR TITLE
Week4

### PR DESCRIPTION
**Pull request by ashwanthdurairaj**
create tic tac toe app in nextjs with typescript

### CODE REVIEW

**Review by DonBrowny — APPROVED** [review]

File: week4/src/pages/pokesearch.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import DisplayTable from './components/displaytable'
+
+function PokeSearch()
+{
+
+    const [search, setSearch] = useState("")
+
+    const [loading, setLoading] = useState(true)
+
+    const [pokemonList, setPokemonList] = useState<Record<string, any>[]>([])
```
Comment (DonBrowny) [inline week4/src/pages/pokesearch.tsx]: you can replace any with a Pokemon interface, create an interface in the schema folder as shown below
```
export interface Pokemon {
  name: string,
  id: number,
  ....
}
```
Then replace any with `Pokemon`
`const [pokemonList, setPokemonList] = useState<Pokemon[]>([])`

File: week4/src/pages/components/displaypokemon.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from "react"
+
+interface pokemonProps{
+    pokemon: Record<string, any>;
```
Comment (DonBrowny) [inline week4/src/pages/components/displaypokemon.tsx]: Avoid using `any`

**Comment by ashwanthdurairaj** [comment]
Created pokedex with svg and pill badges

**Comment by ashwanthdurairaj** [comment]
Add pokemon type filter with minimal design changes

**Pull request merged**